htlcswitch/switch.go

```diff
@@ -704,12 +709,18 @@ func (s *Switch) ForwardPackets(linkQuit <-chan struct{},
 	select {
 	case <-linkQuit:
 		return nil
-	case <-s.quit:
+
+	case <-s.gm.Done():
 		return nil
+
 	default:
 		// Spawn a goroutine to log the errors returned from failed packets.
-		s.wg.Add(1)
-		go s.logFwdErrs(&numSent, &wg, fwdChan)
+		ok := s.gm.Go(func(ctx context.Context) {
+			s.logFwdErrs(ctx, &numSent, &wg, fwdChan)
+		})
+		if !ok {
+			return nil
```

Comment:
There is the following `case` above:
```go
case <-s.gm.Done():
  return nil
```
(It used to be `case <-s.quit` before the PR.)

We get into the branch in question if gm is stopped right before the `Go` call, because if it was stopped before, the above `case` would trigger and we would return `nil`. My reasoning to return `nil` was that the exact timing of gm stopping should not affect the end result.